platform-modules/batch-models/src/main/scala/org/ekstep/analytics/util/StateAdminReportHelper.scala

```diff
@@ -0,0 +1,79 @@
+package org.ekstep.analytics.util
+
+import org.apache.spark.sql.{Encoders, SparkSession}
+import org.apache.spark.sql.functions._
+import org.ekstep.analytics.job.report.StateAdminGeoReportJob.loadData
+import org.ekstep.analytics.job.report.{RootOrgData, SubOrgRow}
+import org.sunbird.cloud.storage.conf.AppConf
+
+class StateAdminReportHelper(sparkSession: SparkSession) {
+  implicit val this.sparkSession = sparkSession
+
+  val tempDir = AppConf.getConfig("admin.metrics.temp.dir")
+  val summaryDir = s"$tempDir/summary"
+
+  def generateGeoBlockData()(implicit sparkSession: SparkSession) = {
+    import sparkSession.implicits._
+    val sunbirdKeyspace = AppConf.getConfig("course.metrics.cassandra.sunbirdKeyspace")
+
+    val locationDF = loadData(sparkSession, Map("table" -> "location", "keyspace" -> sunbirdKeyspace), None).select(
+      col("id").as("locid"),
+      col("code").as("loccode"),
+      col("name").as("locname"),
+      col("parentid").as("locparentid"),
+      col("type").as("loctype"))
+
+    val organisationDF = loadData(sparkSession, Map("table" -> "organisation", "keyspace" -> sunbirdKeyspace), None).select(
+      col("id").as("id"),
+      col("isrootorg").as("isrootorg"),
+      col("rootorgid").as("rootorgid"),
+      col("channel").as("channel"),
+      col("status").as("status"),
+      col("locationid").as("locationid"),
+      col("orgname").as("orgname"),
+      col("locationids").as("locationids"),
+      col("externalid").as("externalid"),
+      col("slug").as("slug")).cache();
```

Comment:
Data loading should be in main job and the data frames then can be passed into methods for further processing.

Reply:
yes, but this particular piece of code will be repeated in 2 jobs. so kept this in a common place.